src/main/java/com/hida/Minter.java

```diff
@@ -0,0 +1,772 @@
+        long uniqueIdCounter = 0;
+
+        // Declares and initializes a list that holds unique values. 
+        // If order matters, tree set is used.
+        Set<Id> uniqueList;
+        if (order) {
+            uniqueList = new TreeSet();
+        } else {
+            uniqueList = new LinkedHashSet(original.size());
+        }
+
+        // iterate through every id 
+        for (Id currentId : original) {
+            // counts the number of times an id has been rejected
+            long counter = 0;
+
+            // continuously increments invalid or non-unique ids
+            while (!DatabaseManager.isValidId(currentId) || uniqueList.contains(currentId)) {
+                /* if counter exceeds totalPermutations, then id has iterated through every 
+                 possible permutation. Related format is updated as a quick look-up reference
+                 with the number of ids that were inadvertedly been created using other formats.
+                 NotEnoughPermutationsException is thrown stating remaining number of ids.
+                 */
+                if (counter > totalPermutations) {
+                    long amountTaken = totalPermutations - uniqueIdCounter;
+
+                    System.out.println("throwing exception");
+                    DatabaseManager.setAmountCreated(
+                            Prefix, TokenType, SansVowel, RootLength, amountTaken);
+                    throw new NotEnoughPermutationsException(uniqueIdCounter, amount);
+                }
+                currentId.incrementId();
+                counter++;
+            }
+            // unique ids are added to list and uniqueIdCounter is incremented.
+            // Size methods aren't used because int is returned
+            uniqueIdCounter++;
+            uniqueList.add(currentId);
+        }
+        System.out.println("\tuniqueCounter = " + uniqueIdCounter);
+        System.out.println("uniqueList size = " + uniqueList.size());
+
+        return uniqueList;
+
+    }
+
+    /**
+     * Generates random ids automatically based on root length. The contents of
+     * the ids are determined by the tokenType.
+     *
+     * @param amount the amount of ids to create
+     * @return a JSON list of unique ids.
+     * @throws SQLException - thrown whenever there is an error with the
+     * database
+     * @throws BadParameterException thrown whenever a malformed or invalid
+     * parameter is passed
+     * @throws java.io.IOException
+     */
+    public String genIdAutoRandom(long amount) throws SQLException, IOException,
+            NotEnoughPermutationsException, BadParameterException {
+        System.out.println("in genIdAutoRandom: " + amount);
+        // checks to see if its possible to produce or add requested amount of
+        // ids to database
+        String tokenMap = BaseMap.get(TokenType);
+
+        Random rng = new Random();

```

Comment:
No need to instantiate one every time, just make it a member variable.  Also, use `SecureRandom` instead